<launch>
  <arg name="tf_prefix" value="$(eval optenv('ROS_NAMESPACE') + '/' if optenv('ROS_NAMESPACE') !=  '' else '')"/>
  <arg name="sim" default="false"/>
  <arg name="wolfgang" default="false"/>
  <arg name="use_game_settings" default="false"/>
  <arg name="no_node" default="false"/>

  <arg if="$(optenv IS_ROBOT false)" name="taskset" default="taskset -c 3"/>
  <arg unless="$(optenv IS_ROBOT false)" name="taskset" default=""/>

  <group if="$(arg sim)">
    <rosparam file="$(find bitbots_quintic_walk)/config/walking_wolfgang_simulator.yaml" command="load"/>
        <group if="$(arg use_game_settings)">
            <rosparam command="load" file="$(find bitbots_bringup)/config/game_settings.yaml" />
        </group>
  </group>
  <group unless="$(arg sim)">
    <rosparam file="$(find bitbots_quintic_walk)/config/walking_wolfgang_robot.yaml" command="load"/>
        <group if="$(arg use_game_settings)">
            <rosparam command="load" file="$(find bitbots_bringup)/config/game_settings.yaml" />
        </group>
  </group>   
  <group unless="$(arg no_node)">
    <node name="walking" pkg="bitbots_quintic_walk" type="WalkNode" output="screen" launch-prefix="$(arg taskset)">
      <param name="base_link_frame" value="$(arg tf_prefix)base_link"/>
      <param name="r_sole_frame" value="$(arg tf_prefix)r_sole"/>
      <param name="l_sole_frame" value="$(arg tf_prefix)l_sole"/>
      <param name="odom_frame" value="$(arg tf_prefix)odom"/>
    </node>
  </group>
</launch>
